massa-proto/proto/massa/api.proto

```diff
@@ -0,0 +1,302 @@
+option php_namespace = "Massa\\Api\\V1";
+
+// Massa gRPC service
+service Grpc {
+    // GetDatastoreEntries
+    rpc GetDatastoreEntries (GetDatastoreEntriesRequest) returns (GetDatastoreEntriesResponse) {
+        option (google.api.http) = {
+            post: "/v1/get_datastore_entries"
+            body: "*"
+        };
+    }
+
+    // GetSelectorDraws
+    rpc GetSelectorDraws (GetSelectorDrawsRequest) returns (GetSelectorDrawsResponse) {
+        option (google.api.http) = {
+            post: "/v1/get_selector_draws"
+            body: "*"
+        };
+    }
+
+    // GetVersion
+    rpc GetVersion (GetVersionRequest) returns (GetVersionResponse) {
+        option (google.api.http) = {get: "/v1/version"};
+    }
+
+    // GetTransactionsThroughput
+    rpc GetTransactionsThroughput (GetTransactionsThroughputRequest) returns (GetTransactionsThroughputResponse) {
+        option (google.api.http) = {get: "/v1/tx"};
+    }
+
+    // GetNextBlockBestParents
+    rpc GetNextBlockBestParents (GetNextBlockBestParentsRequest) returns (GetNextBlockBestParentsResponse) {
+        option (google.api.http) = {get: "/v1/get_next_block_best_parents"};
+    }
+
+    rpc GetBlocksBySlot (GetBlocksBySlotRequest) returns (GetBlocksBySlotResponse) {
+        option (google.api.http) = {get: "/v1/get_blocks_by_slots"};
+    }
+
+
+
+    // ███████╗████████╗██████╗ ███████╗ █████╗ ███╗   ███╗
+    // ██╔════╝╚══██╔══╝██╔══██╗██╔════╝██╔══██╗████╗ ████║
+    // ███████╗   ██║   ██████╔╝█████╗  ███████║██╔████╔██║
+    // ╚════██║   ██║   ██╔══██╗██╔══╝  ██╔══██║██║╚██╔╝██║
+    // ███████║   ██║   ██║  ██║███████╗██║  ██║██║ ╚═╝ ██║
+
+    // SendBlocks
+    rpc SendBlocks (stream SendBlocksRequest) returns (stream SendBlocksResponse) {}
+
+    // SendEndorsements
+    rpc SendEndorsements (stream SendEndorsementsRequest) returns (stream SendEndorsementsResponse) {}
+
+    // SendOperations
+    rpc SendOperations (stream SendOperationsRequest) returns (stream SendOperationsResponse) {}
+
+    // Subscribe GetTransactionsThroughput
+    rpc SubscribeTransactionsThroughput (stream GetTransactionsThroughputStreamRequest) returns (stream GetTransactionsThroughputResponse) {}
+
+    rpc SubscribeNewOperations (stream SubscribeNewOperationsStreamRequest) returns (stream SubscribeNewOperationsStreamResponse) {}
+}
+
+message SubscribeNewOperationsStreamRequest {
+    // id
+    string id = 1;
+
+    repeated OperationStreamFilterType filter = 2;
+}
+
+message SubscribeNewOperationsStreamResponse {
+    string id = 1;
+
+    SecureShareOperation operation = 2;
+}
+
+message GetBlocksBySlotRequest {
+    // id
+    string id = 1;
+
+    // slots
+    repeated Slot slots = 2;
+}
+
+message GetBlocksBySlotResponse {
+    // request id
+    string id = 1;
+
+    // blocks
+    repeated Block blocks = 2;
+}
+
+message GetTransactionsThroughputRequest {
+    //id
+    string id = 1;
+}
+
+message GetTransactionsThroughputStreamRequest {
```

Comment:
Here `interval` still a filter, may be in the future, we would like to add new filters. 

We could keep the same design as `SendBlocks` with a Query struct => Filter struct ?